File: lib/models/pet.dart
// models/pet.dart

class Pet {
  final String id;
  final String name;
  final String breed;
  final String age;
  final String gender;
  final String type;
  final String status;
  final String trait;
  final String imageUrl;

  const Pet({
    required this.id,
    required this.name,
    required this.breed,
    required this.age,
    required this.gender,
    required this.type,
    required this.status,
    required this.trait,
    required this.imageUrl,
  });
}

File: backend/app/Models/Pet.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Pet extends Model
{
    use HasFactory;

    protected $fillable = [
        'name', 'breed', 'age', 'gender', 'type', 
        'status', 'trait', 'image_url'
    ];

    public function adoptions()
    {
        return $this->hasMany(Adoption::class);
    }

    public function favoritedBy()
    {
        return $this->belongsToMany(User::class, 'favorites');
    }
}